core/src/main/scala/org/apache/spark/internal/io/SparkHadoopWriter.scala

```diff
@@ -55,48 +53,61 @@ object SparkHadoopWriter extends Logging {
    * 4. If all tasks are committed, commit the job, otherwise aborts the job;  If any exception is
    *    thrown during job commitment, also aborts the job.
    */
+
   def write[K, V: ClassTag](
       rdd: RDD[(K, V)],
       config: HadoopWriteConfigUtil[K, V]): Unit = {
     // Extract context and configuration from RDD.
     val sparkContext = rdd.context
-    val stageId = rdd.id
-
-    // Set up a job.
-    val jobTrackerId = createJobTrackerID(new Date())
-    val jobContext = config.createJobContext(jobTrackerId, stageId)
-    config.initOutputFormat(jobContext)
-
-    // Assert the output format/key/value class is set in JobConf.
-    config.assertConf(jobContext, rdd.conf)
-
-    val committer = config.createCommitter(stageId)
-    committer.setupJob(jobContext)
-
+    val stageInfo: StageInfo = new StageInfo
+    stageInfo.rddConf = rdd.conf
     // Try to write all RDD partitions as a Hadoop OutputFormat.
     try {
-      val ret = sparkContext.runJob(rdd, (context: TaskContext, iter: Iterator[(K, V)]) => {
+      val ret = sparkContext.runJob(rdd, (stageId: Integer) => {
+        createConf( stageInfo = stageInfo, stageId, config = config)},
```

Comment:
you can simplify how you specify the jobSetupFunc, you can just use `createConf(stageInfo, _, config)`